tgui/packages/tgui/interfaces/common/Objectives.tsx

```diff
@@ -0,0 +1,57 @@
+import { BooleanLike } from 'common/react';
+import { InfernoNode } from 'inferno';
+import { Stack } from '../../components';
+
+export type Objective = {
+  // The title of the objective, not actually displayed so optional
+  name?: string;
+  // What "number" objective this is, IE, its index in the list of objectives
+  count: number;
+  // The text explaining what this objective requires
+  explanation: string;
+  // Whether or not this objective is completed
+  complete: BooleanLike;
+};
+
+export const ObjectivePrintout = (
+  props: {
+    // For passing onto the Stack component
```

Comment:
this is a style nit but I prefer making this an actual type rather than destructuring it here. it makes the function signature easier to read. if you're passionate about this I won't block on it, there's arguments on both sides

Reply:
Considering I didn't know that was a thing until now, I'm not passionate about anything

Reply:
no problem :). i've heard people say destructuring types inline like that makes it more explorable on hover. types can get pretty crazy, so I prefer them as a proper type